src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -193,44 +200,40 @@ public IDataReader ExecuteQuery(string query, CancellationToken cancellationToke
                     }
                 });
 
-                if (numOfQueries == 0 && origReaders.Length > 0)                  // Covers the scenario when user tries to run comments.
+                if (numOfQueries == 0 && origReaders.Length > 0) // Covers the scenario when user tries to run comments.
                 {
-                    var clientRequestProperties = GetCLientRequestProperties(cancellationToken);
+                    var clientRequestProperties = GetClientRequestProperties(cancellationToken);
                     origReaders[0] = _kustoQueryProvider.ExecuteQuery(
-                                KustoQueryUtils.IsClusterLevelQuery(query) ? "" : databaseName,
-                                query,
-                                clientRequestProperties);
+                        KustoQueryUtils.IsClusterLevelQuery(query) ? "" : databaseName,
+                        query,
+                        clientRequestProperties);
                 }
-                
+
                 return new KustoResultsReader(origReaders);
             }
-            catch (AggregateException exception) 
+            catch (AggregateException exception)
                 when (retryCount > 0 &&
-                      exception.InnerException is KustoRequestException innerException 
+                      exception.InnerException is KustoRequestException innerException
                       && innerException.FailureCode == 401) // Unauthorized
             {
                 RefreshAzureToken();
                 retryCount--;
                 return ExecuteQuery(query, cancellationToken, databaseName, retryCount);
             }
         }
-        
+
         /// <summary>
         /// Executes a query or command against a kusto cluster and returns a sequence of result row instances.
         /// </summary>
-        public async Task<IEnumerable<T>> ExecuteControlCommandAsync<T>(string command, bool throwOnError,
+        public async Task ExecuteControlCommandAsync(string command, bool throwOnError,
             CancellationToken cancellationToken)
         {
             try
             {
-                var resultReader = await ExecuteQueryAsync(command, cancellationToken, DatabaseName);
-                var results = KustoDataReaderParser.ParseV1(resultReader, null);
-                var tableReader = results[WellKnownDataSet.PrimaryResult].Single().TableData.CreateDataReader();
-                return new ObjectReader<T>(tableReader);
+                await Task.Run(() => ExecuteControlCommand(command), cancellationToken);
```

Comment:
Changed function to execute command and not execute query since that's the function name.